[READY] Feature/create ahk run configs

takanuva15 opened the pull request:
Hi mtrnord, this PR fixes the run config functionality so that we can run AHK scripts via a run config.  Basically you just create a new AutoHotkey SDK and then set that for the project, and then any scripts that are run will use that SDK.

I have also updated the plugin's compatible build range in the build script to include everything from 2019.3 to 2020.1.  

(Assuming everything looks good, I would like to update the release version to 0.2.0 and push that to JetBrains with this new stuff.)

MTRNord commented on src/main/java/de/nordgedanken/auto_hotkey/sdk/AhkSdkType.java:
```diff
@@ -0,0 +1,86 @@
+
+	@Override
+	public final String getVersionString(String sdkHome) {
+		return "unknown version";
```
Node for the future: Extract it somehow. Maybe using a small ahk script we ship and run.

takanuva15 replied:
Yes, I was thinking the same thing. I should've added a TODO comment with that. I already wrote the script to do it; it's an easy one-liner that prints the version to the console:
`FileAppend %A_AhkVersion%, *`
Then we'd just need to execute it and read the console and save that into the SdkType when the user creates a new Sdk

MTRNord commented:
![grafik](https://user-images.githubusercontent.com/1374914/83296193-9e671480-a1f0-11ea-9f0a-eb222dd1b6b5.png)

I think this is introduced by your PR (I will double-check it). But it seems like the project doesn't have a file tree anymore if I use the open function in the welcome menu (not the import). Instead, it opened a temp file

takanuva15 commented:
> Is it also possible to instead of opening the warning to set the sdk open a dialog to set it? (Not sure if possible but would be nice if it is. Kind of thinking out loud).

I will look into this. There needs to be some kind of error alert shown to the user (otherwise the user has no idea why the Sdk dialog opened); using a message dialog is the easiest way to show it.

Eventually this error code will be deleted in a future release once an Sdk selector is added inside the run config itself (like what the Python plugin has).

takanuva15 commented:
> Is it possible to auto-detect the script path using the open file? I saw other languages do it but I am not sure how.

I am not sure what exactly you mean by this. Are you talking about running a file by clicking the green arrow next to the line numbers within the editor? (Could you give me an example of what steps are being taken?)

---

> I think this is introduced by your PR (I will double-check it). But it seems like the project doesn't have a file tree anymore if I use the open function in the welcome menu (not the import). Instead, it opened a temp file

I checked this against the master branch without my changes, and the same bug is happening. I also checked with a random python file, and the same issue is happening there too:
![image](https://user-images.githubusercontent.com/6986426/83343923-92ea1b00-a2ce-11ea-8912-7cd70c7ef6d7.png)
Thus I think this is an IntelliJ issue. In order to properly open a script, the user should click on the folder they want to open as a project, instead of the script file itself.

---


> Is it also possible to instead of opening the warning to set the sdk open a dialog to set it? (Not sure if possible but would be nice if it is. Kind of thinking out loud).

I have fixed this in my latest commit. If you click "OK" on the warning that shows, it will open up the project settings where you can setup/select the SDK.

As a random note, I also added a small test case to verify how tests work. (Still a lot of documentation to read for that)
